Code/Framework/AzFramework/AzFramework/Spawnable/Spawnable.h

```diff
@@ -29,7 +30,143 @@ namespace AzFramework
         AZ_CLASS_ALLOCATOR(Spawnable, AZ::SystemAllocator, 0);
         AZ_RTTI(AzFramework::Spawnable, "{855E3021-D305-4845-B284-20C3F7FDF16B}", AZ::Data::AssetData);
 
+        // The order is important for sorting in the SpawnableAssetHandler.
+        enum class EntityAliasType : uint8_t
+        {
+            Original,   //!< The original entity is spawned.
+            Disabled,   //!< No entity will be spawned.
+            Replace,    //!< The entity alias is spawned instead of the original.
+            Additional, //!< The original entity is spawned as well as the alias. The alias will get a new entity id.
+            Merge       //!< The original entity is spawned and the components of the alias are added. The caller is responsible for
+                        //!< maintaining a valid component list.
+        };
+
+        enum ShareState : int32_t
+        {
+            Read = -1,
+            NotShared = 0,
+            ReadWrite = 1
+        };
+
+        //! An entity alias redirects the spawning of an entity to another entity, possibly in another spawnable.
+        struct EntityAlias
```

Comment:
To use this it requires typing `Spawnable::EntityAlias` because it's a nested struct, so adding `Spawnable` would just duplicate the same information.

Reply:
That's true. However, I can see some confusion arising when discussing entity aliases for prefabs and spawnables. And I can already see ourselves calling them PrefabEntityAlias and SpawnableEntityAlias to resolve ambiguity during discussions. Hence, my suggestion to name it SpawnableEntityAlias now itself because changing it later will prove challenging since the word 'entityAlias' will be used in many files at that point and we need to be careful where we change the name then.

Reply:
I don't think turning `Spawnable::EntityAlias` into `Spawnable::SpawnableEntityAlias` will change it being referenced as "entity aliases" in discussions rather than "Spawnable entity aliases" and "Prefab entity aliases". There's also not a similar prefix for spawnable entities and prefab entities, which are more prone to confusion imho.